pkg/sql/feature_derivation.go

```diff
@@ -400,5 +401,16 @@ func InferFeatureColumns(trainIR *ir.TrainClause) error {
 			}
 		}
 	}
+	// Write back to client information about feature derivation result
+	if wr != nil {
+		for target, fclist := range trainIR.Features {
```

Comment:
Can we put this code snippet as a separate function `func LogFeatureDerivationResult(wr *PipeWriter, trainIR *ir.TrainClause) error` and call this function at 

https://github.com/sql-machine-learning/sqlflow/blob/a5986ba188feea1081627f32d17383b499f66d35/pkg/sql/submitter.go#L131-L142

so that
1. We can avoid changing `InferFeatureColumns(trainIR)` to `InferFeatureColumns(wr, trainIR)`.
1. We can reuse this function for our debugging due to `w := io.MultiWriter(cw, &output)`.

Reply:
Good point, yet it's better to put it in runSingleSQLIR, where trainIR can be easily got.